src/plugins/intel_npu/src/plugin/npuw/partitioning/partitioning.cpp

```diff
@@ -1557,6 +1560,94 @@ void Partitioner::matchRepeatedSubgraphs(const std::string& func_name) {
     LOG_VERB("Done");
 }
 
+void Partitioner::optimize(const std::string& func_name) {
+    if (!cfg.get<::intel_npu::NPUW_DQ>()) {
+        LOG_VERB("No optimizations will be done to  " << func_name << " in model " << model->get_friendly_name()
+                                                      << "...");
+        return;
+    }
+
+    LOG_VERB("Optimize function " << func_name << " in model " << model->get_friendly_name() << "...");
+    LOG_BLOCK();
+
+    ov::npuw::Function& f = P.functions.at(func_name);
+
+    ov::npuw::patterns::opt::Context ctx;
+    ov::pass::GraphRewrite rewr;
+    rewr.add_matcher<ov::npuw::patterns::opt::DQMatMulCWi>();
+    rewr.add_matcher<ov::npuw::patterns::opt::DQMatMulGQi>(std::ref(ctx));
+    rewr.run_on_model(f._model);
+
+    // Add new parameters where required
+    std::set<ov::npuw::patterns::opt::Context::PPtr> to_delete;
+    std::vector<ov::npuw::patterns::opt::Context::PPtr> new_params;
+    for (auto&& v : ctx.closure_views) {
+        to_delete.insert(v.first.first);
+        new_params.push_back(v.second);
+    }
+    f._model->add_parameters(new_params);
+    ov::pass::Validate().run_on_model(f._model);
+
+    // Transpose tensors where required
+    auto& func_group = all_functions.at(func_name);
+    for (auto&& p : ctx.closures_to_transpose) {
+        auto param_idx = f._model->get_parameter_index(p);
+        auto closure_idx = param_idx - f._param_offset;
+        ov::parallel_for(func_group.refs.size(), [&](std::size_t f_idx) {
+            auto& funcall = func_group.refs[f_idx].get();
+            ov::npuw::util::transpose(funcall._closure[closure_idx]);
+        });
+    }
```

Comment:
Explicit transpose is not required, can be removed, there's a more generic `permute` now.